src/api/Nhs.Appointments.Api/Functions/HttpFunctions/ProposeAvailabilityChangeFunction.cs

```diff
@@ -1,89 +1,91 @@
+
 public class ProposeAvailabilityChangeFunction(
     IBookingAvailabilityStateService bookingAvailabilityStateService,
     IValidator<AvailabilityChangeProposalRequest> validator,
     IUserContextProvider userContextProvider,
     ILogger<ProposeAvailabilityChangeFunction> logger,
     IMetricsRecorder metricsRecorder,
-    IFeatureToggleHelper featureToggleHelper) 
+    IFeatureToggleHelper featureToggleHelper)
     : BaseApiFunction<AvailabilityChangeProposalRequest, AvailabilityChangeProposalResponse>(
-        validator, 
+        validator,
         userContextProvider,
-        logger, 
+        logger,
         metricsRecorder
     )
 {
-
     [OpenApiOperation(operationId: "GetAvailabilityChangeProposal", tags: ["availability"],
         Summary = "Get proposed changes for availability update.")]
-    [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, "application/json", typeof(AvailabilityChangeProposalResponse),
+    [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, "application/json",
+        typeof(AvailabilityChangeProposalResponse),
         Description = "Proposal of how many bookings will be rehomed/unassigned")]
     [OpenApiResponseWithBody(statusCode: HttpStatusCode.Unauthorized, "application/json",
         typeof(ErrorMessageResponseItem), Description = "Unauthorized request to a protected API")]
-    [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.NotFound, Description = "The availability change proposal function is disabled or not available.")]
-    [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.BadRequest, Description = "No matching session was found for the provided request parameters.")]
+    [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.NotFound,
+        Description = "The availability change proposal function is disabled or not available.")]
+    [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.BadRequest,
+        Description = "No matching session was found for the provided request parameters.")]
     [Function("AvailabilityChangeProposalFunction")]
-    public override Task<IActionResult> RunAsync(
+    public override async Task<IActionResult> RunAsync(
         [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "availability/propose-edit")]
         HttpRequest req)
     {
-        return base.RunAsync(req);
+        return await featureToggleHelper.IsFeatureEnabled(Flags.ChangeSessionUpliftedJourney)
+            ? await base.RunAsync(req)
+            : ProblemResponse(HttpStatusCode.NotImplemented, null);
     }
 
-    protected override async Task<ApiResult<AvailabilityChangeProposalResponse>> HandleRequest(AvailabilityChangeProposalRequest request,
+    protected override async Task<ApiResult<AvailabilityChangeProposalResponse>> HandleRequest(
+        AvailabilityChangeProposalRequest request,
         ILogger logger)
     {
-        if (await featureToggleHelper.IsFeatureEnabled(Flags.ChangeSessionUpliftedJourney))
```

Comment:
Are we deliberately dropping the feature flag check here? Is it enforced elsewhere?